### PR TITLE
Add support for auto switch IME between normal mode and insert mode

### PR DESCRIPTION
**Pull request by georgealbert**
This allows VimR to switch IME from a non-ascii input source to  the ascii one when in normal mode, and swich to last input source used in insert mode when enters insert mode.

### CODE REVIEW

**Comment by qvacua** [comment]
Thank you! This has bothered me few times, too, but did not have the chance to find out how to handle input sources. Sorry, it took so long; I've commented few minor things.

**Review by qvacua — CHANGES_REQUESTED** [review]
I've commented two minor things.

File: NvimView/Sources/NvimView/NvimView.swift [hunk 1]
```diff
@@ -297,6 +301,11 @@ public class NvimView: NSView,
 
   var tabEntries = [TabEntry]()
 
+  var asciiImSource : TISInputSource?
+  var lastImSource : TISInputSource?
+
+  var lastMode = ""
```
[the remark this answers is not in the document]
Comment (georgealbert) [reply NvimView/Sources/NvimView/NvimView.swift:307]: I know little of swift. I dont' know how to use CursorModeShape, I use a String for lastMode. We can use CursorModeShape for lastMode

File: NvimView/Sources/NvimView/NvimView.swift [hunk 2]
```diff
@@ -297,6 +301,11 @@ public class NvimView: NSView,
 
   var tabEntries = [TabEntry]()
 
+  var asciiImSource : TISInputSource?
```
[the remark this answers is not in the document]
Comment (georgealbert) [reply NvimView/Sources/NvimView/NvimView.swift:304]: We can initialize both TISInputSource to nil.

File: NvimView/Sources/NvimView/NvimView.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (qvacua) [reply NvimView/Sources/NvimView/NvimView.swift:307]: OK, I'll merge the PR and then slightly refactor. Then, you can comment and/or suggest improvements 😀

**Pull request merged**